src/Generator/ConvertGenerator.php

```diff
@@ -91,10 +90,14 @@ public static function getSupportedTypes() {
 	/** @var EpubGenerator */
 	private $epubGenerator;
 
-	public function __construct( int $timeout, FileCache $fileCache, EpubGenerator $epubGenerator ) {
+	/** @var ?Semaphore */
+	private $semaphore;
+
+	public function __construct( int $timeout, FileCache $fileCache, EpubGenerator $epubGenerator, ?Semaphore $semaphore = null ) {
```

Comment:
Is `$semaphore` able to be null here in order to make the tests easier? Because it looks like in non-test code it never will be.

Reply:
Yes, and to allow people on Windows to bypass the need to provide a semaphore.